### PR TITLE
Initial Waitlisting Feature Dev 

### PR DESCRIPTION
**Pull request by gabrielhan23**
Finished initial backend feature for waitlisting. 
- waitlist model (using timestamps)
- add and drop api for waitlist
- auto add from waitlist after students drop
- light testing with api and admin tools

Not implemented yet:
- coord capabilities
- what to do when student being added from waitlist fails
- waitlist ordering

### CODE REVIEW

File: anonymized_data.json
Comment (smartspot2) [inline anonymized_data.json:1]: This file probably shouldn't be committed

File: docker-compose.yml [hunk 1]
```diff
@@ -33,8 +36,8 @@ services:
         source: ./package-lock.json
         target: /opt/csm_web/package-lock.json
         read_only: true
-      # prevent node modules from being overwritten on host
-      - notused:/opt/csm_web/app/node_modules
+    command: sh -c "npm install && npm start"
+
```
Comment (smartspot2) [inline docker-compose.yml:40]: What's this change for? The entry point should already have done all the setup and server initialization.

(If you revert this, make sure all the changes in this file are reverted.)

**Review by smartspot2 — CHANGES_REQUESTED** [review]
Haven't taken too close of a look at the implementation yet, but just wanted to point out some things I noticed; left some comments about them.

In addition, try to at least get the pytests and the pre-commit check to pass (I don't believe that they'll automatically run on this PR since it's not targeting master, so you may need to manually re-run tests on the PR).

File: csm_web/scheduler/tests/models/test_waitlisted_student.py
Comment (faithdennis) [inline csm_web/scheduler/tests/models/test_waitlisted_student.py:1]: why are we leaving these comments in?

**Review by faithdennis — APPROVED** [review]

**Review by faithdennis — COMMENTED** [review]
Some tests seem to be failing according to the automated tests. Could changing section view have broken other tests? If not, I'm not sure if these automated tests are valid.

**Comment by smartspot2** [comment]
> Some tests seem to be failing according to the automated tests. Could changing section view have broken other tests? If not, I'm not sure if these automated tests are valid.

pre-commit is failing due to the `anonymized_data.json` file being incorrectly committed, and pytest/cypress tests are failing due to the following error:
```
Traceback (most recent call last):

  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute

return self.cursor.execute(sql, params)
psycopg2.errors.NotNullViolation: null value in column "max_waitlist" of relation "scheduler_course" violates not-null constraint

DETAIL:  Failing row contains (1, CS61A, 2024-12-01, 2024-10-02 16:03:44.763658+00, 2024-11-16 17:03:44.763658+00, 2, Structure and Interpretation of Computer Programs, 2024-10-17, f, null, null).
```
Seems like you need to make sure that `max_waitlist` is set for all `Course` models in tests, since it's set as non-null.

File: anonymized_data.json
Comment (gabrielhan23) [reply anonymized_data.json:1]: removed file from commit!

File: docker-compose.yml (hunk 1 above)
Comment (gabrielhan23) [reply docker-compose.yml]: i think it was changed in the very first base commit fixed now!

**Pull request merged**